translation_script/nllb_translate.py

```diff
@@ -0,0 +1,97 @@
+"""
+Script to translate queries using NLLB.
+How to: python nllb_translate.py --queries_file <path to csv/tsv file with queries> --lang <Source language e.g hausa> --output_file <path to output csv/tsv file>
+"""
+import os
+import argparse
+import pandas as pd
+from transformers import AutoModelForSeq2SeqLM, AutoTokenizer
+from transformers import NllbTokenizer
+
+flores_codes = {
+    "bemba" : "bem_Latn",
+    "fon" : "fon_Latn",
+    "hausa" : "hau_Latn",
+    "igbo" : "ibo_Latn",
+    "kinyarwanda" : "kin_Latn",
+    "twi" : "twi_Latn",
+    "yoruba" : "yor_Latn",
+    "swahili" : "swh_Latn",
+    "wolof" : "wol_Latn",
+    "zulu" : "zul_Latn"
+}
+
+def get_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser(
+        description="Translate from Source to Pivot with NLLB"
+    )
+
+    parser.add_argument(
+        "--queries_file",
+        type=str,
+        required=True,
+        help="Path to file containing queries"
+    )
+
+    parser.add_argument(
+        "--lang",
+        type=str,
+        required=True,
+        help="Source language"
+    )
+
+    parser.add_argument(
+        "--output_file",
+        type=str,
+        required=True,
+        help="File path to save translations"
+    )
+
+    return parser
+
+def main():
+    parser = get_parser()
+    args = parser.parse_args()
+
+    assert os.path.exists(args.queries_file), f'The queries file path \{args.queries_file}\ does not exist.'
+
+    if args.queries_file.endswith('csv'):
+        lang_df = pd.read_csv(args.queries_file, sep=",", header=0)
+    elif args.queries_file.endswith('tsv'):
+        lang_df = pd.read_csv(args.queries_file, sep=",", header=0)
+    queries = lang_df['Original question in African language'].to_list()
+
+    tokenizer = AutoTokenizer.from_pretrained("facebook/nllb-200-distilled-600M", 
+        src_lang=flores_codes[args.lang])
+    model = AutoModelForSeq2SeqLM.from_pretrained("facebook/nllb-200-distilled-600M")
+
+    inputs = tokenizer(queries, padding=True, truncation=True, return_tensors="pt")
+
+    print("Translating Queries...")
+    if args.lang == 'fon' or args.lang == 'wolof':
+        translated_tokens = model.generate(
+            **inputs, forced_bos_token_id=tokenizer.lang_code_to_id["fra_Latn"], max_length=30
```

Comment:
I'm just making sure that max_length here does not affect the generations? and does this produce only one translation per question which is what we want

Reply:
Yeah it's one translation per question. 
Hmm, I left it at 30 coz I felt the questions might not have up to 30 tokens? What value can I change it to

Reply:
30 is probably fine, can we test this on once of the language files, probably the one you have access to; it it's okay let's  merge

Reply:
Yup...can I send the result here

Reply:
Naah, if you think it's okay we can merge

Reply:
Yeah it looks fine